### PR TITLE
Update prebuilt rule docs for 8.7

### PR DESCRIPTION
**Pull request by Mikaayenson**
# Summary

Updates the docs per design in https://github.com/elastic/detection-rules/issues/2867#issuecomment-1650370453

### CODE REVIEW

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc [hunk 1]
```diff
@@ -305,7 +301,7 @@ information about a rule's changes, see the rule's description page.
 
 <<startup-or-run-key-registry-modification>>
 
-<<suspicious-.net-code-compilation>>
+<<suspicious-net-code-compilation>>
```
Comment (Mikaayenson) [inline docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:304]: Same thing here. I think we should delete this file altogether right?

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc [hunk 2]
```diff
@@ -5,6 +5,8 @@ The following lists prebuilt rule updates per release. Only rules with
 significant modifications to their query or scope are listed. For detailed
 information about a rule's changes, see the rule's description page.
 
+To see the updates for 8.8 and further releases, please visit the https://www.elastic.co/guide/en/security/8.8/prebuilt-rules-downloadable-updates.html[last version] for the updates.
```
Comment (Mikaayenson) [inline docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:8]:
```suggestion
To see the updates for 8.7 and further releases, please visit the https://www.elastic.co/guide/en/security/8.7/prebuilt-rules-downloadable-updates.html[last version] for the updates.
```
```suggestion
To see the updates for 8.8 and further releases, please visit the https://www.elastic.co/guide/en/security/8.8/prebuilt-rules-downloadable-updates.html[last version] for the updates.
```
```suggestion
To see the updates for 8.8 and further releases, please visit the https://www.elastic.co/guide/en/security/8.8/prebuilt-rules-downloadable-updates.html[last version] for the updates.
```

**Review by approksiu — APPROVED** [review]
Looks good!

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc
Comment (approksiu) [inline docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:1]: keep this page

**Review by approksiu — CHANGES_REQUESTED** [review]
Please keep the "prebuilt rule updates per release" page

**Review by approksiu — APPROVED** [review]

**Review by joepeeples — APPROVED** [review]

**Pull request merged**